File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_email_prefixlength.py
#this is  the email prefix length, it is a spam indicator

#this needs to be modified
def metadata():
    return {  'ApprovedUsage'   : 'ModelAndAllInsights'
            , 'Category'        : 'Lead Information'
            , 'FundamentalType' : 'numeric'
            , 'Description'     : 'Email Prefix Length'
            , 'DisplayName'     : 'Email Prefix Length'
            , 'StatisticalType' : 'ordinal'
            , 'Tags'            : 'Internal' }

def std_visidb_ds_email_prefixlength(email): 
	if '@' not in email:
		return 0
	return email.index('@')    

#function to call
def transform(args, record):
    column = args["column"]
    value = record[column]
    return std_visidb_ds_email_prefixlength(value)

File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsEmailPrefixlength.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.Map;

import com.latticeengines.transform.exposed.RealTimeTransform;
import com.latticeengines.transform.exposed.metadata.ApprovedUsage;
import com.latticeengines.transform.exposed.metadata.Category;
import com.latticeengines.transform.exposed.metadata.FundamentalType;
import com.latticeengines.transform.exposed.metadata.StatisticalType;
import com.latticeengines.transform.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.metadata.TransformMetadata;

public class StdVisidbDsEmailPrefixlength implements RealTimeTransform {

    private static final long serialVersionUID = 664716781231084878L;

    public StdVisidbDsEmailPrefixlength() {
        
    }
    
    public StdVisidbDsEmailPrefixlength(String modelPath) {

    }

    @Override
    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column = (String) arguments.get("column");
        Object o = record.get(column);

        if (o == null)
            return 0;

        String s = (String) o;

        if (s.indexOf("@") < 0)
            return 0;

        return s.indexOf("@");
    }

    @Override
    public TransformMetadata getMetadata() {
        TransformMetadata metadata = new TransformMetadata();
        metadata.setApprovedUsage(ApprovedUsage.MODEL);
        metadata.setCategory(Category.LEAD_INFORMATION);
        metadata.setFundamentalType(FundamentalType.NUMERIC);
        metadata.setStatisticalType(StatisticalType.RATIO);
        metadata.setDescription("Email Prefix Length");
        metadata.setDisplayName("Email Prefix Length");
        metadata.setTags(Tag.INTERNAL_TRANSFORM);
        return metadata;
    }
}
